batteries/cliargs.luau

```diff
@@ -0,0 +1,149 @@
+local cliargs = {}
+cliargs.__index = cliargs
+
+type ArgKind = "positional" | "flag" | "option"
+type ArgOptions = {
+	help: string?,
+	aliases: { string }?,
+	default: string?,
+	required: boolean?,
+}
+
+type ArgData = {
+	name: string,
+	kind: ArgKind,
+	options: ArgOptions,
+}
+
+type ParseResult = {
+	values: { [string]: string },
+	flags: { [string]: boolean },
+	fwdArgs: { string },
+}
+
+type ParseData = {
+	arguments: { [number]: ArgData },
+	positional: { [number] : ArgData},
+	parsed: { [number]: ParseResult },
+}
+
+type ParseInterface = typeof(cliargs)
+export type ParserType = setmetatable<ParseData, ParseInterface>
```

Comment:
```suggestion
export type Parser = setmetatable<ParseData, ParseInterface>
```

The one that is actually exposed should have an "ordinary" type name, don't need to overcomplicate it.